src/ui/ResultFolding/ResultFolding.ts

```diff
@@ -208,9 +208,10 @@ export class ResultFolding extends Component {
       this.getAnalyticsMetadata(),
       this.element
     );
-    this.displayThoseResults(this.result.childResults);
-    this.updateElementVisibility();
-    this.scrollToResultElement();
+    this.displayThoseResults(this.result.childResults).then(() => {
```

Comment:
Nice find that the method is async. I would just suggest using the more modern 'async/await' syntax to handle it. e.g.

```
public async showLessResults() {
    // logic
    await this.displayThoseResults(this.result.childResults);
    // more logic
}
```

Maybe one question, why is `displayThoseResults` async? and is there a chance it could fail?

Reply:
Ok, I also prefer async/await
For `displayThoseResults` It waits till all results are rendered and appended to the DOM correctly

Reply:
But my question is since the rest of the file uses `Promise` and `then`, wouldn't it be confusing to use two way of doing async?
Is it worth is to go ahead and refactor all of it?

Reply:
I think it's okay if both live side-by-side. My view is as we work on a piece of code, we always want to update to the latest standards/syntax etc.

With changes like `var` -> `const` or `function()` -> `() => {}`, I feel okay doing an entire file at once. With async code however, in my experience, unexpected bugs can be introduced if refactoring to async/await quickly.

I'm okay with you changing the rest of the file, but I would ask to test the changes to make sure no bad behaviour gets introduced. I'm also okay with gradually updating the code as we work on different bugs/jiras.